lib/account/account-service.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+/**
+ * What we initially send back to the client so they can start to initiate
+ * the complete procedure and save information.
+ *
+ * @typedef {accountService~initiateResult}
+ * @param {string} accountId
+ * @param {string} email
+ * @param {string} mfaKey
+ * @param {string} salt
+ */
+
+/**
+ * Handles account creation from initiating to confirmation.
+ */
+class AccountService {
+    /**
+     * Used for Dependency Injection and initial configuration
+     * of the storage plugin.
+     *
+     * @param {Object} config
+     * @param {Object} password
+     * @param {Object} storage
+     */
+    constructor(config, password, storage) {
+        this.accountFile = null;
+        this.password = password;
+        this.storage = storage.configure(config.storage);
+    }
+
+
+    /**
+     * Completes the sign up of the account with more information from the
+     * client. We need the client to create the password using their tools
+     * so we never see the plain text version of it. This helps to keep
+     * the system more secure
+     *
+     * TODO: Encrypt data
+     *
+     * @param {string} directory
+     * @param {Object} accountInfo
+     * @param {Object} options
+     * @return {Promise.<boolean>}
+     */
+    completeAsync(directory, accountInfo, options) {
+        this.accountFile.password = accountInfo.password;
+
+        return this.storage.putAsync(directory, JSON.stringify(this.accountFile), options).then(() => {
```

Comment:
Do a "get" and remove the caching mechanism you have for now.  We should have a cache that works throughout the lifetime of a request and is cleared after each request.  That's harder to do, so let's take the performance penalty of two S3 gets.